interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,97 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', ['Genesets', function(Genesets) {
+  var genesets = [];
+  var query = {};
+  var resultCount = null;
+
+  return {
+    getQuery: function() {
+      return query;
+    },
+    getGenesets: function() {
+      return genesets;
+    },
+    clear: function() {
+      query = {};
+      genesets = [];
+    },
+    getResultCount: function() {
+      return resultCount;
+    },
+    query: function(searchParams) {
+      query = searchParams;
+      Genesets.query(query, function(data) {
+        resultCount = data.meta.total_count;
+        genesets = data.objects;
+      });
+    }
+  };
+}])
+
+.directive('genesetSearchForm', ['GenesetSearch', function(GenesetSearch) {
```

Comment:
'GenesetSearch' no longer needs to be injected here. You aren't using it in the controller and it's not referenced in the template.